public/app/core/logs_model.ts

```diff
@@ -311,6 +314,8 @@ export function logSeriesToLogsModel(logSeries: DataFrame[]): LogsModel | undefi
     for (let j = 0; j < series.length; j++) {
       const ts = timeField.values.get(j);
       const time = dateTime(ts);
+      const tsNs = datasourceId === 'loki' ? timeNanosecondField.values.get(j) : undefined;
+      const timeEpochNs = Number(datasourceId === 'loki' ? tsNs : time.valueOf() + '000000');
```

Comment:
I don't think we need the datasource here at all, we can just check if there is a `tsNs` field and use it otherwise just fallback to miliseconds.